src/middlewared/middlewared/alert/source/snapshot_count.py

```diff
@@ -17,37 +17,48 @@ class SnapshotCountAlertClass(AlertClass):
     level = AlertLevel.WARNING
     title = "Too Many Snapshots Exist For Dataset"
     text = (
-        "Dataset %(dataset)s has more snapshots (%(count)d) than recommended (%(max)d). Performance or functionality "
-        "might degrade."
+        "SMB share %(dataset)s has more snapshots (%(count)d) than recommended (%(max)d). File Explorer may not "
+        "display all snapshots in the Previous Versions tab."
     )
 
 
 class SnapshotCountAlertSource(AlertSource):
     schedule = CrontabSchedule(hour=1)
     run_on_backup_node = False
 
-    async def check(self):
-        max_ = await self.middleware.call("pool.snapshottask.max_count")
+    async def _check_total(self, snapshot_counts: dict[str, int]) -> list[Alert]:
+        """Return an `Alert` if the total number of snapshots exceeds the limit."""
         max_total = await self.middleware.call("pool.snapshottask.max_total_count")
-
-        total = 0
-        datasets = await self.middleware.call("zfs.snapshot.count")
-
-        for cnt in datasets.values():
-            total += cnt
+        total = sum(snapshot_counts.values())
 
         if total > max_total:
-            return Alert(
+            return [Alert(
                 SnapshotTotalCountAlertClass,
                 {"count": total, "max": max_total},
                 key=None,
-            )
+            )]
+
+        return []
 
-        for dataset in sorted(datasets.keys()):
-            count = datasets[dataset]
+    async def _check_smb(self, snapshot_counts: dict[str, int]) -> list[Alert]:
+        """Return an `Alert` for every dataset shared over smb whose number of snapshots exceeds the limit."""
+        max_ = await self.middleware.call("pool.snapshottask.max_count")
+        to_alert = list()
+
+        for share in await self.middleware.call("sharing.smb.query"):
+            path = share["path"].removeprefix("/mnt/")
```

Comment:
Share path may begin with `EXTERNAL:` in which case we should _not_ get snapshots.

Reply:
You can import `path_location` method from `middlewared.utils.path` and then check the location type and if it is not FSLocation.LOCAL skip the check.

Reply:
Should we also exclude external snapshot counts from `_check_total`?